nix/esy/default.nix

```diff
@@ -0,0 +1,289 @@
+    src = builtins.fetchTarball {
+      url = https://gforge.inria.fr/frs/download.php/36602/cudf-0.9.tar.gz;
+      sha256 = "12p8aap34qsg1hcjkm79ak3n4b8fm79iwapi1jzjpw32jhwn6863";
+    };
+    buildPhase = ''
+      make all opt
+    '';
+    patchPhase = "sed -i s@/usr/@$out/@ Makefile.config";
+    createFindlibDestdir = true;
+  };
+
+  dose3 = stdenv.mkDerivation {
+    name = "dose";
+    src = builtins.fetchurl {
+      url = "http://gforge.inria.fr/frs/download.php/file/36063/dose3-5.0.1.tar.gz";
+      sha256 = "00yvyfm4j423zqndvgc1ycnmiffaa2l9ab40cyg23pf51qmzk2jm";
+    };
+    buildInputs = with esyOcamlPkgs; [
+      ocaml
+      findlib
+      ocamlbuild
+      cppo
+      perl
+    ];
+    propagatedBuildInputs = with esyOcamlPkgs; [
+      cudf
+      ocaml_extlib
+      ocamlgraph
+      re
+    ];
+    createFindlibDestdir = true;
+    patches = [
+      ./patches/0001-Install-mli-cmx-etc.patch
+      ./patches/0002-dont-make-printconf.patch
+      ./patches/0003-Fix-for-ocaml-4.06.patch
+      ./patches/0004-Add-unix-as-dependency-to-dose3.common-in-META.in.patch
+      ./patches/dose.diff
+    ];
+  };
+
+  esy-solve-cudf = esyOcamlPkgs.buildDunePackage rec {
+    pname = "esy-solve-cudf";
+    version = "0.1.10";
+    buildInputs = with esyOcamlPkgs; [
+      ocaml
+      findlib
+      dune
+    ];
+    propagatedBuildInputs = with esyOcamlPkgs; [
+      cmdliner
+      cudf
+      ocaml_extlib
+    ];
+    src = fetchFromGitHub {
+      owner  = "andreypopp";
+      repo   = pname;
+      rev    = "v${version}";
+      sha256 = "174q1wkr31dn8vsvnlj4hzfgvbamqq74n7wxhbccriqmv8lz5a3g";
+      fetchSubmodules = true;
+    };
+
+    buildPhase = ''
+      runHook preBuild
+      dune build -p ${pname},mccs
+      runHook postBuild
+    '';
+
+    meta = {
+      homepage = https://github.com/andreypopp/esy-solve-cudf;
+      description = "package.json workflow for native development with Reason/OCaml";
+      license = stdenv.lib.licenses.gpl3;
+    };
+  };
+
+  # XXX(anmonteiro): The NPM registry doesn't allow us to fetch version
+  # information for scoped packages, and `@esy-nightly/esy` is scoped. It also
+  # seems that Esy only uses the `package.json` file to display the version
+  # information in `esy --version`, so we can kinda ignore this for now. We're
+  # able to build and install nightly releases but it'll always display the
+  # current version information.
+  esyNpm = builtins.fetchurl {
+    url = "https://registry.npmjs.org/esy/${esyVersion}";
+    sha256 = "0rhbbg7rav68z5xwppx1ni8gjm6pcqf564nn1z6yrag3wgjgs63c";
+  };
+
+  esySolveCudfNpm = builtins.fetchurl {
+    url = "https://registry.npmjs.org/esy-solve-cudf/${esy-solve-cudf.version}";
+    sha256 = "19m793mydd8gcgw1mbn7pd8fw2rhnd00k5wpa4qkx8a3zn6crjjf";
+  };
+
+in
+  esyOcamlPkgs.buildDunePackage rec {
+    pname = "esy";
+    version = esyVersion;
+
+    minimumOCamlVersion = "4.06";
+
+    src = fetchFromGitHub {
```

Comment:
I get an error where I try to build `src = ../..` with it: `Error: Library "rely.lib" not found.`. How can I add rely?

Reply:
I think you wanna grab a commit after https://github.com/esy/esy/pull/1006

Reply:
It works, thx.